management-gateway/src/main/java/org/frankframework/management/gateway/SerializableInputStream.java

```diff
@@ -0,0 +1,181 @@
+		}
+	}
+
+	/**
+	 * Default constructor, to wrap the original stream in.
+	 */
+	public SerializableInputStream(final InputStream in) {
+		super();
+
+		LOG.trace("creating new SerializableInputStream with delegate");
+		delegate = in;
+		tmpFile = null;
+	}
+
+	@Override
+	public void writeExternal(ObjectOutput out) throws IOException {
+		LOG.trace("Writing delegate to ObjectOutput with buffersize [{}]", BUFFERSIZE);
+
+		try(InputStream is = delegate) {
+			int overallBytes = 0;
+			final byte[] buffer = new byte[BUFFERSIZE];
+			int bytesRead;
+			while ((bytesRead = is.read(buffer, 0, BUFFERSIZE)) > 0) {
+				LOG.trace("wrote [{}] bytes to ObjectOutput", bytesRead);
+				out.writeInt(bytesRead);
+				out.write(buffer, 0, bytesRead);
+				overallBytes += bytesRead;
+			}
+			out.writeInt(0);
+			LOG.trace("finished serializing stream, total bytes written [{}]", overallBytes);
+		} finally {
+			delegate = null;
+		}
+	}
+
+	@Override
+	public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+		try (OutputStream fos = Files.newOutputStream(tmpFile)) {
+			copyStream(in, fos);
+		}
+
+		delegate = Files.newInputStream(tmpFile);
+	}
+
+	private void copyStream(ObjectInput in, OutputStream fos) throws IOException {
+		int overallBytes = 0;
+		int toRead = in.readInt();
+		overallBytes += toRead;
+		byte[] buffer = null;
+		while (toRead > 0) {
+			LOG.trace("reading up to [{}] bytes from ObjectInput", toRead);
+			buffer = allocateBuffer(toRead, buffer);
+			final int readLocally = in.read(buffer, 0, toRead);
+			LOG.trace("read [{}] bytes from stream and stored them in the buffer", readLocally);
+			fos.write(buffer, 0, readLocally);
+
+			toRead -= readLocally;
+			if (toRead == 0) {
+				toRead = in.readInt();
+				overallBytes += toRead;
+			}
+		}
+		LOG.trace("finished deserializing stream, total bytes read [{}]", overallBytes);
+	}
+
+	/**
+	 * Initially creates a buffer equal to the length of the first bytes 'part', set by {@link SerializableInputStream#BUFFERSIZE}.
+	 */
+	private byte[] allocateBuffer(final int size, final byte[] buffer) {
+		if (buffer != null && buffer.length >= size) {
+			return buffer;
+		}
+		return new byte[size];
+	}
+
+	@Override
+	public int read() throws IOException {
+		checkIfDelegateExists();
+
+		return delegate.read();
+	}
+
+	@Override
+	public int read(byte[] b) throws IOException {
+		checkIfDelegateExists();
+
+		return delegate.read(b);
+	}
+
+	@Override
+	public int read(byte[] b, int off, int len) throws IOException {
+		checkIfDelegateExists();
+
+		return delegate.read(b, off, len);
+	}
+
+	private void checkIfDelegateExists() throws IOException {
+		if(delegate == null) {
+			throw new IOException("delegate has not been set, unable to read stream");
+		}
+	}
+
+	@Override
+	public void close() throws IOException {
+		try {
+			if(delegate != null) {
+				delegate.close();
+			}
```

Comment:
`CloseUtils.silentClose()`?

Reply:
Haha, I'll merge the master so I can use this instead 😄